openquake/hazardlib/gsim/zhao_2016.py

```diff
@@ -439,11 +601,24 @@ class ZhaoEtAl2016Asc(GMPE):
 
     #: Required rupture parameters are magnitude, top-of-rupture depth and
     #: style of faulting (rake)
-    REQUIRES_RUPTURE_PARAMETERS = {'mag', 'rake', 'ztor'}
+    REQUIRES_RUPTURE_PARAMETERS = {'mag', 'ztor', 'rake'}
 
     #: Required distance measure is Rrup and Rvolc
     REQUIRES_DISTANCES = {'rrup', 'rvolc'}
 
+    def __init__(self, volc_arc_file=None, **kwargs):
+        super().__init__(volc_arc_file=volc_arc_file, **kwargs)
+        
+        if volc_arc_file is not None:
+            with open(volc_arc_file, 'rb') as fle:
+                self.volc_arc_file = fle.read().decode('utf-8')
```

Comment:
`volc_arc_file` is a bad name, since this is not a file, it is the content of the underlying file. I would rename it to
`volc_arc_str` or something like that.

Reply:
Changed to volc_arc_str